### ISSUE TITLE
Duplicates in associations table

### ISSUE THREAD

**Issue by curtchan** [DATA-DOG/DataDogAuditBundle#60]
I don't think it should create multiple rows with same data whenever object is used?

**Comment by l3pp4rd** [comment]
well, this is an audit log, would you expect to lose all history if association object is removed?

**Comment by curtchan** [comment]
> well, this is an audit log, would you expect to lose all history if association object is removed?

You misunderstood me. The issue is that it creates multiple rows for same entity - that is same tbl, same entity, same fkey, i highly doubt if that's supposed to happen like that.
Example:
Someone updates entity A 3 times.
log table has 3 rows - nice.
association log table has... 6 rows, where 4 are duplicates, what's the point of that?

**Comment by l3pp4rd** [comment]
Yes probably if association remains unchanged it could use the most recent version of it.

**Comment by indjeto** [comment]
I agree with that.
Now it creates a new entries in the associations every time, so if intended to work this way, then better this data to be part of the audit_log table.
I have some ideas for optimizations, can I contribute here, or the project is frozen?

**Comment by l3pp4rd** [comment]
sure you can contribute, since I no longer use PHP there is noone to maintain it. will add you to maintainer list
